res/controllers/Pioneer-DDJ-ERGO-scripts.js

```diff
@@ -0,0 +1,419 @@
+        const deck = status-0x90;
+        if (engine.getParameter(group, "play") === 1) {
+            engine.setParameter(group, "cue_gotoandstop", 1);
+        } else {
+            engine.setParameter(group, "cue_gotoandplay", 1);
+        }
+        //store sample numbers
+        selectedSamples[0] = (control - 0x3C)/2 + (deck)*4;
+        selectedSamples[1] = (control - 0x3C)/2 + (deck)*4;
+    }
+};
+
+//this modifies the selected sample volume with sample vol knob rotations
+DDJERGO.samplerVolume = function(channel, control, value, status) {
+    const deck = status-0xB0;
+    let newValue;
+    if (value-64 > 0) { newValue = value-128; } else { newValue = value; }
+    if (newValue > 0) {
+        script.triggerControl(samplerStrings[selectedSamples[deck]], "pregain_up");
+    }
+    if (newValue < 0) {
+        script.triggerControl(samplerStrings[selectedSamples[deck]], "pregain_down");
+    }
+};
+
+//this plays the selected sample with a sample vol knob click
+DDJERGO.samplerKnobClick = function(channel, control, value, status) {
+    const deck = status-0x90;
+    const sample = selectedSamples[deck];
+    if (value === 0x7F) {
+        if (engine.getParameter(samplerStrings[sample], "play") === 1) {
+            engine.setParameter(samplerStrings[sample], "cue_gotoandstop", 1);
+        } else {
+            engine.setParameter(samplerStrings[sample], "cue_gotoandplay", 1);
+        }
+    }
+};
+
+//this changes the sample stored in the knob being shift + turned
+DDJERGO.rotateSamples = function(channel, control, value, status) {
+    const deck = status-0xB0;
+    let newValue;
+    if (value-64 > 0) { newValue = value-128; } else { newValue = value; }
+    selectedSamples[deck] += newValue;
+    //makes sure samples stay within range
+    if (selectedSamples[deck] < 0) {
+        selectedSamples[deck] += 8;
+    }
+    if (selectedSamples[deck] > 7) {
+        selectedSamples[deck] -= 8;
+    }
+};
+
+//this changes both the mic and aux pregains with the aux/mic vol knob
+DDJERGO.micAuxGain = function(channel, control, value) {
+    //weird non-linear mapping for pregain
+    if (value < 64) {
+        value = value/63;
+    } else {
+        value = 3*value/64 - 125/64;
+    }
+    engine.setValue("[Microphone]", "pregain", value);
+    engine.setValue("[Auxiliary1]", "pregain", value);
+};
+
+//this toggles on/off effect 1 or 2 based on FX [1-2] buttons clicks
+DDJERGO.quickEffects = function(channel, control, value, status) {
+    if (value === 0x7F) {
+        const deck = control % 2;
+        const effect = (control - deck - 0x4C)/4;
+        const toggleState = 1 - engine.getParameter(effectsStrings[deck][effect], "enabled");
+        engine.setParameter(effectsStrings[deck][effect], "enabled", toggleState);
+        //LEDs for FX number button and FX [1-2]
+        midi.sendShortMsg(0x94 + deck, 0x47 + effect, 0x7F*toggleState);
+        midi.sendShortMsg(status, control, 0x7F*toggleState);
+    }
+};
```

Comment:
The components Fx mapping now takes care of 1-2-3 knops and button.
What does this do?
Is it for the Fx 1-2 buttons?
If yes, those should toggle the deck -> Fx unit assignment. https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[EffectRack1_EffectUnitN]-group_[ChannelI]_enable
Left deck:
(1) `[EffectRack1_EffectUnit1],group_[Channel1]_enable`
(2) `[EffectRack1_EffectUnit2],group_[Channel1]_enable`
Right deck:
same, just with Channel2